LipidDyn/core.py

```diff
@@ -334,73 +335,144 @@ def __init__ (self,
         -------------
         universe : object 
             MDAnalysis universe object
-        selection : object
-            MDAnalysis AtomGroups selections   
+        ncore : int 
+            Number of cores employed for the calculation    
         """
         self.universe = universe
-        self.selection = selection
+        self.bin_size = bin_size
+        self.ncore = ncore 
 
-        bins = 0.02
         ts = self.universe.trajectory[0] 
         # divide by ten the coordinates to convert in nm 
-        self.n1 =  int(round((ts.dimensions[0]*0.1)/bins)) # X coordinate of box
-        self.n2 =  int(round((ts.dimensions[1]*0.1)/bins)) # Y coordinate of box
+        self.n1 =  int(round((ts.dimensions[0]*0.1)/self.bin_size)) # X coordinate of box
+        self.n2 =  int(round((ts.dimensions[1]*0.1)/self.bin_size)) # Y coordinate of box
        
 
-    def do2dgrid(self,
-                 trajectory,
-                 l_grids):
+    def multiprocessing(self,
+                        selection):
+        
+        """Function to parallelize and speed up the 
+           calculations of the 2D density maps.
 
-        """Execute the membranes command of fatslim,
-        which identifies the upper and lower leaflet of
-        a membrane simulation.
+
+            Parameters
+            -------------
+            selection : MDAnalysis AtomGroup
+
+            Return:
+            ------------
+            grid: numpy.array
+                Numpy array containig the normalized density 
+                values
+        """
+
+        u = self.universe
+        ncore = self.ncore
+
+        # divide the frames in chunk      
+        chunks= np.linspace(0, 
+                            len(u.trajectory)-1, 
```

Comment:
can you actually check if this `-1` is required? I did some further testing and I think it might lead us to skip the last frame. Please double check